cmd/lifecycle/analyzer.go

```diff
@@ -88,8 +89,22 @@ func (a *analyzeCmd) Args(nargs int, args []string) error {
 		}
 	}
 
+	targetRegistry, err := parseRegistry(a.outputImageRef)
+	if err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse target registry")
+	}
+
 	if a.previousImageRef == "" {
 		a.previousImageRef = a.outputImageRef
+	} else if !a.useDaemon {
+		previousRegistry, err := parseRegistry(a.previousImageRef)
+		if err != nil {
+			return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse previous registry")
+		}
+		if previousRegistry != targetRegistry {
```

Comment:
Why do we need to fail in this case? I'm thinking about a situation where the platform decides to move registries from ECR to GGCR.

Reply:
Same answer as here. The failure was always happening in the exporter but I moved it up in the spirit of "failing fast".